### DISCUSSION TITLE
REAL-TIME DATA AUTOMATION!!!

### DISCUSSION THREAD

**Discussion by saent-x** [DaveSkender/Stock.Indicators#792] [Ideas]
Hello Everyone!! I'm new here.
I'd like to know if I can use this library with a real-time market data connection, if so how do I go about it?? For instance if I want to use this library to connect with a brokers love market data to trade real-time how do I go about it?? Thanks

**Comment by DaveSkender** [comment]
First, determine which data provider you want to use.  Here are [some suggestions](https://github.com/DaveSkender/Stock.Indicators/discussions/579).  Next, get the data from your provider and put it into a [quote format](https://daveskender.github.io/Stock.Indicators/guide/#historical-quotes).  Finally, [use the library](https://daveskender.github.io/Stock.Indicators/guide/#example-usage) to generate indicators.  There is also a [complete example code project](https://daveskender.github.io/Stock.Indicators/examples/#content) you can download to see and learn from.

**Reply by saent-x** [reply]
Okay so this library is strictly for backtesting right?

**Reply by DaveSkender** [reply]
No, not at all.  You can use this data in any way you can imagine.  With that said, the actual function of this library is to simply generate indicator data from standard OHLCV format financial market data.

> Send in historical price quotes and get back desired technical indicators.  Nothing more.

Here are some things you can build with the resulting indicator data, to name a few:

- algo trading systems
- strategy backtesting engines
- market event/alert notifications
- general market analysis
- machine-learning analysis
- charting systems ([example](https://stock-charts.azurewebsites.net/))

Here are some examples of what others are doing:
https://github.com/DaveSkender/Stock.Indicators/network/dependents

**Reply by kiranmaya** [reply]
Im using for both backtesting and runtime. You should convert your candle sticks to this library . just make a new quote list from your source.let me know if you want to see how i did it. Ill post the code.

**Reply by NavjotSinghMinhas** [reply]
@DaveSkender In case of real-time scenario. Fetching an indicator value for **N** Quotes will result in IndicatorResult. But what if I want to add one more quote (**N+1**) to get the indicator of the new quote? 

As I understand from the code is I have to pass the whole N+1 list for the calculation for just 1 new Quote?

**Reply by DaveSkender** [reply]
@NavjotSinghMinhas Currently, yes, you have to provide enough quote history to cover the lookback periods needed to calculate the indicator.  Most of these indicators cannot be calculated from just one candle in isolation without the history.

With that said, I will be adding features to support incremental streaming data in v2, after I finish a few things to complete v1.  See [the product roadmap](https://github.com/DaveSkender/Stock.Indicators/discussions/218).  I can get to these features faster with more help from the community … [help wanted!](https://github.com/DaveSkender/Stock.Indicators/blob/main/docs/contributing.md#readme)

**Comment by DaveSkender** [comment]
Follow or continue this conversation in  our [discussion on streaming indicators](https://github.com/DaveSkender/Stock.Indicators/discussions/1018) for progress and information about pre-release package versions to support live and incremental quote sources.
